src/System.Text.Json/tests/JsonElementWriteTests.cs

```diff
@@ -339,6 +339,24 @@ public static void WriteSimpleObject(bool indented)
                 "{\"r\":2,\"d\":2}");
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public static void WriteSimpleObjectNeedsEscaping(bool indented)
+        {
+            WriteComplexValue(
+                indented,
+                @"{ ""prop><erty""   : 3,
+            ""> This is one long & unusual property name. <"":
+4
+}",
+                @"{
+  ""prop\u003e\u003certy"": 3,
+  ""\u003e This is one long \u0026 unusual property name. \u003c"": 4
+}",
+                "{\"prop\\u003e\\u003certy\":3,\"\\u003e This is one long \\u0026 unusual property name. \\u003c\":4}");
```

Comment:
nit: Consider using upper case for the hex digits (a-f -> A-F)

The existing tests have lower case sprinkled around because that was the previous expected output. But since we use JavascriptEncoder now, the output is upper case hex digits. For new tests, lets match the correct casing and reduce reliance on this helper:
https://github.com/dotnet/corefx/blob/e6a5951fb68a09598f1fc1826a94de212385c649/src/System.Text.Json/tests/Utf8JsonWriterTests.cs#L5904-L5906